### PR TITLE
gh-109276: regrtest: shorter list of resources

### PR DESCRIPTION
**Pull request by vstinner**
* Issue: gh-109276

### CODE REVIEW

**Comment by vstinner** [comment]
Examples:

```
$ ./python -m test -u network
== resources (1): network

$ ./python -m test --slow-ci
== resources: all

$ ./python -m test --fast-ci
== resources: all, -cpu


$ ./python -m test -u all,tzdata
== resources: all, +tzdata

$ ./python -m test 
== resources: all test resources are disabled, use -u option to unskip tests
```

**Comment by vstinner** [comment]
@serhiy-storchaka: Does it look better like that?

File: Lib/test/libregrtest/utils.py [hunk 1]
```diff
@@ -535,6 +548,31 @@ def is_cross_compiled():
...
 
 
+def format_resources(use_resources: tuple[str, ...]):
+    # set preserves insertion order
+    use_resources = set(use_resources)
+    all_resources = set(ALL_RESOURCES)
+
+    # Express resources relative to "all"
+    relative_all = ['all']
+    for name in all_resources - use_resources:
```
Comment (serhiy-storchaka) [inline Lib/test/libregrtest/utils.py]: Use `sorted()` for deterministic result.

**Review by serhiy-storchaka — COMMENTED** [review]
I did not have problems with the older output, but if you want to change it, the new one looks better.

Did you consider the idea of making it compatible with the input to the `-u` option? I.e. remove space after comma and remove plus sign. Alternatively you can make `-u` accepting input with spaces and pluses (although copying text with spaces is less convenient). Not that I personally need this feature, but it might come in handy one day.

**Comment by vstinner** [comment]
> Did you consider the idea of making it compatible with the input to the -u option? I.e. remove space after comma and remove plus sign.

Done.

> Use sorted() for deterministic result.

Done.

File: Lib/test/libregrtest/utils.py [hunk 2]
```diff
@@ -535,6 +548,30 @@ def is_cross_compiled():
...
+def format_resources(use_resources: tuple[str, ...]):
+    use_resources = set(use_resources)
+    all_resources = set(ALL_RESOURCES)
+
+    # Express resources relative to "all"
+    relative_all = ['all']
+    for name in sorted(all_resources - use_resources):
+        relative_all.append(f'-{name}')
+    for name in sorted(use_resources - all_resources):
+        relative_all.append(f'{name}')
```
Comment (serhiy-storchaka) [inline Lib/test/libregrtest/utils.py:560]:
```suggestion
        relative_all.append(name)
```

**Review by serhiy-storchaka — APPROVED** [review]

File: Lib/test/libregrtest/utils.py (hunk 2 above)
Comment (vstinner) [reply Lib/test/libregrtest/utils.py:560]: Oh right. Sadly, I saw your comment after the change was merged. I make try to include it in my next regrtest change ;-)

**Comment by vstinner** [comment]
> I did not have problems with the older output

For me, it's not easy to know if it's default --slow-ci option, default --fast-ci option, or if -u options were modified on a buildbot. With the new output relative to ``-u all``, it should be easier to guess the test config.

**Pull request merged**